File: burstLibrary/src/main/java/com/harry1453/burst/explorer/entity/Transaction.java
package com.harry1453.burst.explorer.entity;

import java.math.BigInteger;

public class Transaction {
    public final BurstValue amount;
    public final BigInteger blockID;
    public final String fullHash;
    public final BigInteger confirmations;
    public final BurstValue fee;
    public final BigInteger type;
    public final String signatureHash;
    public final String signature;
    public final BurstAddress sender;
    public final BurstAddress recipient;
    public final BigInteger timestamp;
    public final BigInteger transactionID;
    // todo attachment
    public final BigInteger subType;

    public Transaction(BurstValue amount, BigInteger blockID, String fullHash, BigInteger confirmations, BurstValue fee, BigInteger type, String signatureHash, String signature, BurstAddress sender, BurstAddress recipient, BigInteger timestamp, BigInteger transactionID, BigInteger subType) {
        this.amount = amount;
        this.blockID = blockID;
        this.fullHash = fullHash;
        this.confirmations = confirmations;
        this.fee = fee;
        this.type = type;
        this.signatureHash = signatureHash;
        this.signature = signature;
        this.sender = sender;
        this.recipient = recipient;
        this.timestamp = timestamp;
        this.transactionID = transactionID;
        this.subType = subType;
    }
}

File: burstLibrary/src/main/java/com/harry1453/burst/explorer/entity/Transaction.kt
package com.harry1453.burst.explorer.entity

import java.math.BigInteger

data class Transaction (
        @JvmField val amount: BurstValue,
        @JvmField val blockID: BigInteger,
        @JvmField val fullHash: String,
        @JvmField val confirmations: BigInteger,
        @JvmField val fee: BurstValue,
        @JvmField val type: BigInteger,
        @JvmField val signatureHash: String,
        @JvmField val signature: String,
        @JvmField val sender: BurstAddress,
        @JvmField val recipient: BurstAddress,
        @JvmField val timestamp: BigInteger,
        @JvmField val transactionID: BigInteger,
        // TODO Attachment
        @JvmField val subType: BigInteger
)